src/types.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Type for when we can take a CSS Selector, or an HTML Element (mostly mounting).
+ */
+export type ElementOrSelector = keyof HTMLElementTagNameMap | HTMLElement;
```

Comment:
This file contains all the types used in the project. There are places where type definitions will be inferred, or would not be trivial to add here. Where possible I've included them here, and will probably add more as they become necessary.